tests/integration/_common.py

```diff
@@ -232,44 +287,84 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
+
+            showing = self._pane.capture_pane()
+
+            if showing:
+                if wait_on_cli_prompt:
+                    # Wait for the cli prompt at the end, a stdout test
+                    if len(showing) > 1:
+                        ok_to_return.append(self._cli_prompt in showing[-1])
+                    else:
+                        ok_to_return.append(False)
                 else:
-                    ok_to_return.append(False)
+                    # Guarantee that the screen has changed
+                    ok_to_return.append(showing != self._last_screen)
+                    if showing:
+                        # Guarantee navigator is up and running
+                        ok_to_return.append(":help help" in showing[-1])
+                        # Wait for a specific playbook status
+                        if wait_on_playbook_status:
+                            ok_to_return.append(showing[-1].endswith(wait_on_playbook_status))
+                        # Wait for the lack of "Collecting collection content"
+                        if wait_on_collection_fetch_prompt:
+                            ok_to_return.append(wait_on_collection_fetch_prompt not in showing[0])
             else:
-                if wait_on_help:
-                    ok_to_return.append(any(":help help" in line for line in showing))
-                if wait_on_playbook_status:
-                    ok_to_return.append(showing[-1].endswith(wait_on_playbook_status))
-                if wait_on_collection_fetch_prompt:
-                    ok_to_return.append(wait_on_collection_fetch_prompt not in showing[0])
+                ok_to_return.append(False)
+
+            if all(ok_to_return):
+                break
+
+            elapsed = timer() - start_time
+            if elapsed > timeout:
+                with open(setup_capture_path, "w") as filehandle:
+                    filehandle.writelines("\n".join(self._setup_capture))
+
+                tstamp = datetime.datetime.now(tz=datetime.timezone.utc).isoformat()
+                # taint the screen output w/ timestamp so it's never a valid fixture
+                alert = f"******** ERROR: TMUX TIMEOUT @ {elapsed}s @ {tstamp} ********"
+                showing.insert(0, alert)
+                with open(timeout_capture_path, "w") as filehandle:
+                    filehandle.writelines("\n".join(showing))
+                self._fail_remaining = ["******** Previous test failure ********"]
+                return showing
+
+        self._last_screen = showing
         return showing
 
     def _get_cli_prompt(self):
         """get cli prompt"""
+        # give tmux a second to start
+        # start a fresh clean shell, set TERM
+        self._pane.send_keys("env -i bash --noprofile --norc")
+        self._pane.send_keys("export TERM=xterm")
         self._pane.send_keys("clear")
+        bash_prompt_visible = False
         showing = []
-        while len(showing) != 1:
+        while True:
```

Comment:
Can we add another timer here? Otherwise if we get here and have an issue, we stall CI for an hour again.

Reply:
got it